core/embed/extmod/modtrezorcrypto/modtrezorcrypto-secp256k1_zkp.h

```diff
@@ -325,13 +331,265 @@ STATIC mp_obj_t mod_trezorcrypto_secp256k1_context_multiply(
+    mp_raise_ValueError("Invalid length of asset");
+  }
+  mp_buffer_info_t blind;
+  mp_get_buffer_raise(blind_obj, &blind, MP_BUFFER_READ);
+  if (blind.len != 32) {
+    mp_raise_ValueError("Invalid length of blinding factor");
+  }
+  secp256k1_generator gen;
+  if (!secp256k1_generator_generate_blinded(ctx, &gen, asset.buf, blind.buf)) {
+    mp_raise_ValueError("Generator blinding failed");
+  }
+  byte out[33] = {0};
+  secp256k1_generator_serialize(ctx, out, &gen);
   return mp_obj_new_bytes(out, sizeof(out));
 }
+
 STATIC MP_DEFINE_CONST_FUN_OBJ_3(
-    mod_trezorcrypto_secp256k1_context_multiply_obj,
-    mod_trezorcrypto_secp256k1_context_multiply);
+    mod_trezorcrypto_secp256k1_context_blind_generator_obj,
+    mod_trezorcrypto_secp256k1_context_blind_generator);
+
+// TODO: this code is copied from modtrezorcrypto-monero.h - find a better way
+// to handle the conversion.
+static uint64_t _mp_obj_get_uint64(mp_const_obj_t arg) {
+  if (MP_OBJ_IS_SMALL_INT(arg)) {
+    return MP_OBJ_SMALL_INT_VALUE(arg);
+  } else if (MP_OBJ_IS_TYPE(arg, &mp_type_int)) {
+    byte buff[8];
+    uint64_t res = 0;
+    mp_obj_t *o = MP_OBJ_TO_PTR(arg);
+
+#if MICROPY_LONGINT_IMPL != MICROPY_LONGINT_IMPL_MPZ
+#error "MPZ supported only"
+#endif
+    mp_obj_int_to_bytes_impl(o, true, 8, buff);
+    for (int i = 0; i < 8; i++) {
+      res <<= i > 0 ? 8 : 0;
+      res |= (uint64_t)(buff[i] & 0xff);
+    }
+    return res;
+  } else {
+    if (MICROPY_ERROR_REPORTING == MICROPY_ERROR_REPORTING_TERSE) {
+      mp_raise_TypeError("can't convert to int");
+    } else {
+      nlr_raise(mp_obj_new_exception_msg_varg(&mp_type_TypeError,
+                                              "can't convert %s to int",
+                                              mp_obj_get_type_str(arg)));
+    }
+  }
+}
+
+/// def pedersen_commit(self, value: long, blinding_factor: bytes, gen: bytes)
+/// -> bytes:
+///     '''
+///     Commit to specified integer value, using given 32-byte blinding factor.
+///     '''
+STATIC mp_obj_t mod_trezorcrypto_secp256k1_context_pedersen_commit(
+    size_t n_args, const mp_obj_t *args) {
+  const secp256k1_context *ctx =
+      mod_trezorcrypto_get_secp256k1_context(args[0]);
+
+  const uint64_t value = _mp_obj_get_uint64(args[1]);
+
+  mp_buffer_info_t blind;
+  mp_get_buffer_raise(args[2], &blind, MP_BUFFER_READ);
+  if (blind.len != 32) {
+    mp_raise_ValueError("Invalid length of blinding factor");
+  }
+  mp_buffer_info_t gen;
+  mp_get_buffer_raise(args[3], &gen, MP_BUFFER_READ);
+  if (gen.len != 33) {
+    mp_raise_ValueError("Invalid length of generator");
+  }
+  secp256k1_generator generator;
+  if (!secp256k1_generator_parse(ctx, &generator, gen.buf)) {
+    mp_raise_ValueError("Generator parsing failed");
+  }
+
+  secp256k1_pedersen_commitment commit;
+  if (!secp256k1_pedersen_commit(ctx, &commit, blind.buf, value, &generator)) {
+    mp_raise_ValueError("Pedersen commit failed");
+  }
+
+  byte output[33] = {0};
```

Comment:
nit: no need to initialize to 0. (don't know what the code style is)

Reply:
Removed the initialization.